scripts/dumps/dumpGenomeFromCore.pl

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env perl
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+use strict;
+use warnings;
+
+
+use Bio::EnsEMBL::PaddedSlice;
+use Bio::EnsEMBL::Utils::IO::FASTASerializer;
+use Bio::EnsEMBL::DBSQL::DBAdaptor;
+
+use Getopt::Long;
+
+my $dbname;
+my $host;
+my $port;
+my $mask;
+my $genome_dump_file;
+my $help;
+my $desc = "
+This script dump all toplevel sequences from a core database and store it at the fatsa.
+The sequences can be unmasked, soft masked or hard masked.
+
+USAGE dumpGenomeFromCore.pl [-mask] -coreDB COREDB -host HOST -port PORT -outfile OUTFILE
+
+Options:
+* -coreDB
+      the core database storing the genome sequence to be dumped
+* --host
+      server hosting the core database
+* --port
+      port for the host database
+* --outfile
+      file where the dumped seqeunce will be sored at the fasta format
+* --mask
+      level of masking of the dumped sequences [soft/hard]. If this option is not defined then the
+      sequence will be unmasked.
+";
+
+GetOptions(
+    'coreDB=s' => \$dbname,
+    'host=s'   => \$host,
+    'port=s'     => \$port,
+    'mask=s'     => \$mask,
+    'outfile=s'  => \$genome_dump_file,
+    'help' => \$help
+  );
+
+
+if ($help) {
+  print $desc;
+  exit(0);
+}
+
+my $dba = Bio::EnsEMBL::DBSQL::DBAdaptor->new( -user   => 'ensro',
+                                               -dbname => $dbname,
+                                               -host   => $host,
+                                               -port   => $port,
+                                               -driver => 'mysql');
+
+my $slice_adaptor = $dba->get_adaptor("Slice");
+my $slices = $slice_adaptor->fetch_all("toplevel");
```

Comment:
Alternatively:
```suggestion
my $slices = $dba->get_SliceAdaptor->fetch_all("toplevel");
```
Just to make sure, you don't want non-reference slices, duplicates or LRGs, right?

Reply:
yes we just want reference slices